### ISSUE TITLE
Bugs still ingame

### ISSUE THREAD

**Issue by ByBurton** [Sh0wdown/UnofficialCrusaderPatch#5]
I just felt like doing a list of bugs / quirky stuff I still know are a problem ingame, and I will list all I remember from the top of my head here.

- [ ] European bowmen won't set pitch alight
- [x] AI will not erase tower ruins
- [x] AI will not build more than 3 tower mangonels / ballistae or trebuchets / Fire ballistae
- [ ] Wolf won't man his towers with Arabic Bowmen
- [x] AI erases stables if they are not accessible
- [x] AI deletes all buildings outside its castle if not accessible (gate closed) sometimes.
- [x] Abbot and Emir do not buy bows
- [x] AI sets their bakeries to sleep if no flour is present in the stockpile. Even though some are still producing bread. Same with any weapon production or mills, or breweries and inns.
- [ ] Rat is too bad :( Please buff my poor ratty :(

I wanted to add, that I am royally impressed with your accomplishments, and I could not wait for the next update, so I installed Visual Studio and compiled it myself ;) 

Edit:
Maybe I will also add a few things I feel like mentioning:

- [x] The AI should put more units on their towers / walls (I am looking at you, Rat, Richard)
- [ ] The Rat should build a few more apple farms, or something, so he can actually earn some money. Would go a long way. I think the Rat is just a little too weak...
- [ ] The Sultan should rebuild firethrowers. And also have some defensive archers next to his attacking ones.
- [ ] Some AI opponents should have the number for attacker for their first attack set to something higher, because they attack very late anyways (Wolf for example)

Edit2:
Another bug:
- [ ] The AI sometimes attacks the stockpiles for stone next to a quarry. And these sometimes stay, even if the corresponding quarry was destroyed.

**Comment by Heroesflorian** [comment]
> The AI should put more units on their towers / walls (I am looking at you, Rat, Richard)

Not to forget everyone's favourite, King Phillip! :D

> The Sultan should rebuild firethrowers. 

Oh yes! And also Saladin and some others who seem to slack in that regard... 

> Wolf won't man his towers with Arabic Bowmen

Similarly, Caliph also doesn't use slingers atop his walls. And possibly there's even more such cases hidden in the AIV files. 

> AI will not build more than 3 tower mangonels / ballistae or trebuchets / Fire ballistae

While I definitely agree it's a bit sad that one couldn't even make Richard-style square castle with all mangonels/ballista with the limit of 3, I would guess an AI castle with 5 tower ballistae would be OP and they capped it at 3 due to that, considering that most AI characters do not make a lot of siege engines for their attacks either. Hence, I personally feel those two things should be changed "together", if possible.

**Comment by ByBurton** [comment]
> While I definitely agree it's a bit sad that one couldn't even make Richard-style square castle with all mangonels/ballista with the limit of 3, I would guess an AI castle with 5 tower ballistae would be OP and they capped it at 3 due to that, considering that most AI characters do not make a lot of siege engines for their attacks either. Hence, I personally feel those two things should be changed "together", if possible.


Yes, it would make enemies stronger, but if we address their weak attacks, by strengthening their manpower, we can also strengthen the defences. Also, most aiv's show 4 siege engines / tower engines, which makes sense, because of mirroring. It just looks stupid seeing e.g. Wolf8.aiv with two fire ballistae on the right tower, but only one on the left....

**Comment by Heroesflorian** [comment]
> Yes, it would make enemies stronger, but if we address their weak attacks, by strengthening their manpower, we can also strengthen the defences. Also, most aiv's show 4 siege engines / tower engines, which makes sense, because of mirroring. It just looks stupid seeing e.g. Wolf8.aiv with two fire ballistae on the right tower, but only one on the left....

Yep. 

--

Also, regarding the "known bugs" list... I got some more ones to add to that: 
- [ ] AI deletes some or all buildings even INSIDE its castle despite being accessible (no gate, walls or other obstacles between keep, campfire and affected buildings) sometimes. (Note: Whenever buildings inside/outside the castle are deleted by bug, the AI does not get back any resources, which adds to the severity!)

- [ ] AI instantly "disappears" all soldiers, workers and peasants inside and outside the castle, including those stationed on towers, etc. for no apparent reason sometimes. This does sometimes but not always include tower-mounted siege engines as well. Accessibility, unit limits, popularity, fire, plague, opponent attacks, lions and finances are all unrelated to this behaviour (yes, this is a really mysterious and scary one!). 

- [ ] When removing engineers from a siege engine while at or above the per-player unit limit, those engineers often either disappear or turn into hostile workers or soldiers.

**Comment by ByBurton** [comment]
I could also add another bug:

- [

**Comment by MaCl00** [comment]
And all ranged units seem to miss when aiming at a unit which is moving through bog (Pech).

**Comment by MaCl00** [comment]
>  AI deletes all buildings outside its castle if not accessible (gate closed) sometimes. 

This happens especially when the AI is using drawbridges: Every time they are *opened*, the AI treats the drawbridge as moat and thinks that all the outer buildings are not accessible and deletes them.

**Comment by Panbutt** [comment]
I would like to add to this, some AI bugs I've noticed with Phillip, Rat, Pig and Richard.

-Phillip will never re-train tower crew archers that have been killed.
-Rat, Pig and Richard are the only Lords that do not build moats, despite that they have tons of moat-digging units.
-Pig builds one Pitch Rig, however never uses pitch and sells the pitch immediately when gathered. Allowing him to hang into at least 5 pitch before selling would allow him to use Oil Smelters effectively. It would also let him use defensive pitch as well. Pig is all about brutality, it makes sense for him to use boiling oil.

I also want to say thank you for the community patch, it has pumped new life into Crusader and I love all the changes made so far <3.

**Comment by PitchNeeded** [comment]
> -Pig will never sell food. Some of his AIVs have 2 granaries which means he can stockpile up to 500 cheese. Allowing him to sell food would give him a good economic boost.

I wondered why Pig often seems to have so much cheese in his granary! Thanks for solving this mystery! Allowing him to sell food would be excellent for his economy. It means he would only need one granary too, so one of his granaries also could be removed and replaced with a fletchers workshop in the vanilla aiv patch!

**Comment by MaCl00** [comment]
>  It seems to vary greatly in frequency even between multiple runs of an identical matchup.

Yes I noticed the same thing. But I start to get the feeling that it happens more often with the mod on "high-speed" (eg 200+)...

**Comment by brooklynzsmac** [comment]
Sultan doesn't produce any food.

**Comment by PitchNeeded** [comment]
> Sultan doesn't produce any food.

I wouldn't really call that a bug, its just part of his character, and kind of works for Sultan as he has a low population so any food he buys isn't eaten that quickly, and also he can focus on just defending his quarries which he does rather well (although he could do with sending a few more Arabian swordsmen to defend them), rather than having to defend farms too. I think therefore Sultan would benefit more from another quarry than farms.

**Comment by Lolasik011** [comment]
Same bug... Pig. 
![pig bug](https://user-images.githubusercontent.com/41998977/48576921-d49eb500-e926-11e8-8d5f-897f9117fefc.png)

**Comment by ByBurton** [comment]
Poor piggy :(

**Comment by PitchNeeded** [comment]
> Poor piggy :(

Indeed. I never knew about this bug and so it was a mystery and rather frustrating when during a skirmish match a certain AI would completely stop recruiting troops and start accumulating loads of gold instead.

**Comment by Lolasik011** [comment]
Is it possible to fix it ?

**Comment by Sh0wdown** [comment]
> Is it possible to fix it ?

If you are referring to the ox tether spam, that will be fixed in the next release.

**Comment by Heroesflorian** [comment]
> If you are referring to the ox tether spam, that will be fixed in the next release.

That sounds great!

**Comment by PitchNeeded** [comment]
Another bug that I noticed:

Often an AI's last stand defence mode (where they send a message expressing their worry about the attack, sleep all their buildings and try to recruit troops) isn't triggered by macemen, even when an overwhelming number of macemen attack them and are outside their castle attacking walls/towers for a while before they kill them. I have noticed this with Emir, Richard and Wazir, they often don't seem to react when I attack one of their towers with loads of macemen to get into the castle.

**Comment by TheSnowyChickens** [comment]
Question:
Is it possible for you to fix graphic bugs? 
Like: 
- "The Wall in the Gate/Tower" 
- "The Stone (the highest Stone) (from Quarries, in the little stockpile) half chopped off, because the Quarry himself"
- "The 1 block bigger hitbox from Apple farms" (in the under construction preview)

**Comment by PitchNeeded** [comment]
Two more bugs:
1. AI raiding troops sometimes wait in the field, or wait on aggressive mode, for new reinforcements to come rather than moving to their next target.
2. AI mangonels attack deceased lord statues

**Comment by AdmiralRigby** [comment]
One important bug: Fire ballistas don't attack monks on their own. Even when given the command to attack a monk, they won't attack others after the selected one has fallen.

**Comment by PitchNeeded** [comment]
Also from my own observations, mangonels cannot seem to do any damage to fire ballistas.

**Comment by ByBurton** [comment]
> One important bug: Fire ballistas don't attack monks on their own. Even when given the command to attack a monk, they won't attack others after the selected one has fallen.

They also do not attack the Lord.

**Comment by Heroesflorian** [comment]
@PitchNeeded mangonels can deal damage to fire ballistae though. They are not particularly accurate, but they fire at hostile ballistae in range, and when the rocks hit, they properly deal damage. But yeah, they're not particularly great weapons against single/few units in general, due to their bad accuracy, they're more of a crowd control weapon.

**Comment by Sh0wdown** [comment]
Closing this as most of them are fixed or fixable through AIC edits.
If I missed something please create a new issue with one problem per issue.
